core/lib/merkle_tree/src/lib.rs

```diff
@@ -123,12 +123,12 @@ use zksync_crypto::hasher::blake2::Blake2Hasher;
 ///
 /// [Jellyfish Merkle tree]: https://developers.diem.com/papers/jellyfish-merkle-tree/2021-01-14.pdf
 #[derive(Debug)]
-pub struct MerkleTree<'a, DB> {
+pub struct MerkleTree<DB> {
     db: DB,
-    hasher: &'a dyn HashTree,
+    hasher: &'static dyn HashTree,
```

Comment:
At this point is there any advantage of keeping a reference to `dyn HashTree` vs having `MerkleTree` generic over `HashTree` and storing it directly?

Reply:
Dynamic dispatch slightly reduces compilation times and produced executable size at the cost of theoretical performance degradation (I haven't observed any noticeable performance degradation when performing benchmarks locally.) Since we have a single hash function when using Merkle tree, a parametric implementation should produce that much overhead in practice, so I think switching to it makes sense.